Fix updating for excerpt from content in Google preview

hannaw93 opened the pull request:
## Context
<!--
What do we want to achieve with this PR? Why did we write this code?
-->

* When changing the content of an existing post, the excerpt in the Google preview does not get updated unless the page is refreshed. The PR fixes this problem.

## Summary

<!--
Attach one of the following labels to the PR: `changelog: bugfix`, `changelog: enhancement`, `changelog: other`, `changelog: non-user-facing`.
If the changelog item is a bugfix, please use the following sentence structure: Fixes a bug where ... would ... (when ...).
If the changelog item is meant for the changelog of another add-on, start your changelog item with the name of that add-on's repo between square brackets, for example: * [wordpress-seo-premium] Fixes a bug where ....
If the changelog items is meant for the changelog of a javascript package, specify between square brackets in which package changelog the item should be included, for example: * [@yoast/components] Fixes a bug where ....
If the same changelog item is applicable to multiple changelogs/add-ons, add a separate changelog item for all of them.
-->
This PR can be summarized in the following changelog entry:

* Fixes updating of the excerpt from post content in the Google preview

## Relevant technical choices:
* Retrieving excerpt from content has been covered in `configurations.js` so it was removed from `dom.js`.
* The helper for` gets the term excerpt from the document` was removed from `dom.js` since the excerpt functionality is not available for taxonomies.

## Test instructions
<!--
Please follow these guidelines when creating test instructions:
- Please provide step-by-step instructions how to reproduce the issue, if applicable.
- Write step-by-step test instructions aimed at non-tech-savvy users, even if the PR is not user-facing.
-->
### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Create a post 
* Add the %%excerpt%% replacement variable to meta description in the Google preview snippet
* Make sure an excerpt from the beginning of the post appears in the Google preview metadescription
* Save the post
* Go to the posts overview and reopen the created post
* Add a text to the beginning of the post
* Make sure the Google preview metadescription immediately gets updated

Additional steps:
* Create a separate excerpt field through navigating to `screen options` on the `edit post` page and checking the box `excerpt`
* Add an excerpt to the field
* Make sure the added excerpt appears in the Google preview

### Test instructions for QA when the code is in the RC
<!--
Sometimes some steps from the test instructions for the acceptance test aren't relevant anymore once the code has been merged or the feature is complete. If that is the case, do not check the checkbox below.
QA is our Quality Assurance team. The RC is the release candidate zip that is tested before a release 
-->

* [x] QA should use the same steps as above.

<!--
If the above checkbox has not been checked, write down all steps QA should take to test this PR, not only the difference with the acceptance test steps. If QA should use the test instructions specified on the epic, paste a link to the relevant comment on the epic.
-->
QA can test this PR by following these steps:

*

## Impact check
<!--
Sometimes PRs have a bigger impact than is suggested in the user-facing changes. In such cases,
additional (regression) testing might be necessary. To make it clear what parts might need additional testing, please outline which parts of the plugin have been impacted by this PR.
-->
This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes

* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Other environments

* [ ] This PR also affects Shopify. I have added a changelog entry starting with `[shopify-seo]` and I have added test instructions for Shopify.

## Documentation

* [ ] I have written documentation for this change.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended
* [ ] If any part of the code is behind a feature flag, my test instructions also cover cases where the feature flag is switched off.

Fixes #https://yoast.atlassian.net/jira/software/c/projects/LINGO/boards/99?modal=detail&selectedIssue=LINGO-1311

agnieszkaszuba commented:
CR: needs changes

When I go on a category or tag page, the metabox doesn't appear. This gets fixed when I add back this line that was removed: `export const getTermExcerpt = () => excerptFromContent( getTermDescription(), getMetaDescriptionLimit() );` 

Maybe the problem occurs because we removed the declaration of getTermExcerpt but there are still usages of it that haven't been removed, for example in packages/js/src/classic-editor/initial-state.js

agnieszkaszuba commented:
I'm actually also not sure what is meant by "The helper for gets the term excerpt from the document was removed from dom.js since the excerpt functionality is not available for taxonomies."

There is no excerpt box for taxonomies, but it's still possible to add an excerpt from content. When we remove the line in dom.js I don't know if it will be possible anymore.

hannaw93 commented:
> CR: needs changes
> 
> When I go on a category or tag page, the metabox doesn't appear. This gets fixed when I add back this line that was removed: `export const getTermExcerpt = () => excerptFromContent( getTermDescription(), getMetaDescriptionLimit() );`
> 
> Maybe the problem occurs because we removed the declaration of getTermExcerpt but there are still usages of it that haven't been removed, for example in packages/js/src/classic-editor/initial-state.js

Thanks! I removed it's usage from initial-state.js and tests :)

hansjovis reviewed (approved):
CR: 👍 . 

I did [transform a comment into a doc block](https://github.com/Yoast/wordpress-seo/pull/18096/commits/176afa11ef32aedb7ae6112c445c08d341cab503), though 🙂 .

hansjovis commented:
Acceptance test: ✅
